File: ruby/2.rb
#!/usr/bin/ruby

sum = 0
n = 1
lastn = 0
while n < 4000000 do
	newn = n + lastn
	if newn % 2 == 0 then
		sum += newn
	end
	lastn = n
	n = newn
end

puts(sum)

File: php/2.php
#!/usr/bin/php
<?php

$sum = 0;
$n = 1;
$lastn = 0;
while ($n < 4000000) {
	$newn = $lastn + $n;
	if ($newn % 2 == 0)
		$sum += $newn;
	$lastn = $n;
	$n = $newn;
}
echo $sum . PHP_EOL;

?>